sidecar/src/config.rs

```diff
@@ -126,16 +129,13 @@ impl FromEnv {
                 println!("help: {ENV_SIDECAR_LOG_METHOD}: {SIDECAR_LOG_METHOD_DISABLED}|{SIDECAR_LOG_METHOD_STDOUT}|{SIDECAR_LOG_METHOD_STDERR}|file:///path/to/file");
                 LogMethod::default()
             }
-            _ => parse_uri(method.as_str())
-                .ok()
-                .and_then(|u| {
-                    if Some("file") == u.scheme_str() {
-                        Some(LogMethod::File(PathBuf::from(u.path())))
-                    } else {
-                        None
-                    }
-                })
-                .unwrap_or_default(),
+            method if method.starts_with("file:///") => {
+                let fixed_uri = format!("file://./{}", &method[7..]);
+                Uri::from_str(&fixed_uri)
+                    .map(|u| LogMethod::File(PathBuf::from(&u.path())))
+                    .unwrap_or_default()
+            }
```

Comment:
Why not just this instead of the roundtrip to url?
```suggestion
            method if method.starts_with("file://") => {
                let path = method.strip_prefix("file://").unwrap();
                LogMethod::File(PathBuf::from(path))
            }
```

Reply:
It's because the specification for the path is in the form `file:///my/path`. However, even though this is a valid URI (with an empty host), see RFC 3986:

```
  URI         = scheme ":" hier-part [ "?" query ] [ "#" fragment ]

      hier-part   = "//" authority path-abempty
                  / path-absolute
                  / path-rootless
                  / path-empty

      authority   = [ userinfo "@" ] host [ ":" port ]

      host        = IP-literal / IPv4address / reg-name
      reg-name    = *( unreserved / pct-encoded / sub-delims )

   If the URI scheme defines a default for host, then that default
   applies when the host subcomponent is undefined or when the
   registered name is empty (zero length).  For example, the "file" URI
   scheme is defined so that no authority, an empty host, and
   "localhost" all mean the end-user's machine, whereas the "http"
   scheme considers a missing authority or empty host invalid.
```

the http package rejects the uri for not having a host. So add a bogus host (.).

Reply:
Going through the `Uri` struct is not a good idea. If there are non-alphanumeric characters, they would need to be URL-encoded for this to work. This value comes from an env variable that can be manually supplied so encoding is painful to do.

We already do not follow the URI standard for unix sockets URL (since the logic is just `unix:// + path on disk without any special encoding` ) so I don't see the point of arbitrarily following it for this.

Reply:
ah sorry I didn't read that right. I can't just strip the prefix because it may have urlencoded characters